app/src/main/java/org/connectbot/ConsoleActivity.java

```diff
@@ -669,255 +661,86 @@ public boolean onTouch(View v, MotionEvent event) {
 		if (tabs != null)
 			setupTabLayoutWithViewPager();
 
-		// detect fling gestures to switch between terminals
-		final GestureDetector detect = new GestureDetector(this, new GestureDetector.SimpleOnGestureListener() {
-			private float totalY = 0;
-
+		pager.setOnClickListener(new OnClickListener() {
 			@Override
-			public void onLongPress(MotionEvent e) {
-				super.onLongPress(e);
-				openContextMenu(pager);
-			}
-
-
-			@Override
-			public boolean onScroll(MotionEvent e1, MotionEvent e2, float distanceX, float distanceY) {
-
-				// if copying, then ignore
-				if (copySource != null && copySource.isSelectingForCopy())
-					return false;
-
-				if (e1 == null || e2 == null)
-					return false;
-
-				// if releasing then reset total scroll
-				if (e2.getAction() == MotionEvent.ACTION_UP) {
-					totalY = 0;
-				}
-
-				// activate consider if within x tolerance
-				int touchSlop = ViewConfiguration.get(ConsoleActivity.this).getScaledTouchSlop();
-				if (Math.abs(e1.getX() - e2.getX()) < touchSlop * 4) {
-
-					View view = adapter.getCurrentTerminalView();
-					if (view == null) return false;
-					TerminalView terminal = (TerminalView) view;
-
-					// estimate how many rows we have scrolled through
-					// accumulate distance that doesn't trigger immediate scroll
-					totalY += distanceY;
-					final int moved = (int) (totalY / terminal.bridge.charHeight);
-
-					// consume as scrollback only if towards right half of screen
-					if (e2.getX() > view.getWidth() / 2) {
-						if (moved != 0) {
-							int base = terminal.bridge.buffer.getWindowBase();
-							terminal.bridge.buffer.setWindowBase(base + moved);
-							totalY = 0;
-							return true;
-						}
-					} else {
-						// otherwise consume as pgup/pgdown for every 5 lines
-						if (moved > 5) {
-							((vt320) terminal.bridge.buffer).keyPressed(vt320.KEY_PAGE_DOWN, ' ', 0);
-							terminal.bridge.tryKeyVibrate();
-							totalY = 0;
-							return true;
-						} else if (moved < -5) {
-							((vt320) terminal.bridge.buffer).keyPressed(vt320.KEY_PAGE_UP, ' ', 0);
-							terminal.bridge.tryKeyVibrate();
-							totalY = 0;
-							return true;
-						}
-
-					}
-
+			public void onClick(View v) {
+				if (keyboardGroup.getVisibility() == View.GONE) {
+					showEmulatedKeys(false);
 				}
-
-				return false;
 			}
-
-
 		});
 
-		pager.setLongClickable(true);
-		pager.setOnTouchListener(new OnTouchListener() {
+		if (Build.VERSION.SDK_INT < Build.VERSION_CODES.HONEYCOMB) {
+			pager.setOnTouchListener(new OnTouchListener() {
```

Comment:
Would it be possible to add this listener in TerminalView instead? Then we would have both the pre- and post-Honeycomb code in the same spot. It's confusing to have the two different copy implementations spread throughout different files.